lib/services/appmanager-service.ts

```diff
@@ -50,14 +52,52 @@ class AppManagerService implements IAppManagerService {
 			let projectName = this.$project.projectData.ProjectName;
 			let solutionPath = buildResult[0].solutionPath;
 			let projectPath = solutionPath.substr(solutionPath.indexOf("/") + 1);
-			this.$server.tam.uploadApplication(projectName, projectName, projectPath).wait();
 
-			this.$logger.info("Successfully uploaded package.");
+			let publishSettings: Server.PublishSettings = {
+				IsPublished: options["publish"],
+				NotifyByPush: options["send-push"],
```

Comment:
it is better to use options.sendPush
As `send-push` option has dash ("-"), yargs and our validation will allow passing --sendPush option. In case you use `options["send-push"]` and user specifies --sendPush, you will not receive the data passed by him.
`--sendPush` guarantees you that you will always receive correct data.